taxi/views.py

```diff
@@ -61,10 +67,34 @@ class CarListView(LoginRequiredMixin, generic.ListView):
 class CarDetailView(LoginRequiredMixin, generic.DetailView):
     model = Car
 
+    def post(self, request, *args, **kwargs):
+        action = request.POST.get("action")
+        pk = kwargs["pk"]
```

Comment:
as an alternative you can use `self.get_object()` - on DetailView it will pull the instance that we are working with so you will need to pass pk ina form